src/map/unit.cpp

```diff
@@ -717,15 +718,17 @@ int unit_walktoxy( struct block_list *bl, short x, short y, unsigned char flag)
 	if (!path_search(&wpd, bl->m, bl->x, bl->y, x, y, flag&1, CELL_CHKNOPASS)) // Count walk path cells
 		return 0;
 
+	if (bl->type != BL_NPC) {
+
 #ifdef OFFICIAL_WALKPATH
-	if( !path_search_long(NULL, bl->m, bl->x, bl->y, x, y, CELL_CHKNOPASS) // Check if there is an obstacle between
-		&& wpd.path_len > 14	// Official number of walkable cells is 14 if and only if there is an obstacle between. [malufett]
-		&& (bl->type != BL_NPC) ) // If type is a NPC, please disregard.
-			return 0;
+		if( !path_search_long(NULL, bl->m, bl->x, bl->y, x, y, CELL_CHKNOPASS) // Check if there is an obstacle between
```

Comment:
Hi, random passerby here, if you're going for optimization, you might want to chain these ifs together into a single if:
if (not an npc **AND** (there is an obstacle **OR** the path exceeds the maximum length))
This could technically be faster than nested ifs.

Reply:
Right, thanks for pointing it out!